### DISCUSSION TITLE
Confusing warning "ignoring invalid defined png:compression-filter =5"

### DISCUSSION THREAD

**Discussion by uprprc777** [ImageMagick/ImageMagick#7236] [Help] [answered]
### ImageMagick version

7.1.1-30

### Operating system, version and so on

macOS 14.4.1

### Description

Hello!

I'm attempting to use ImageMagick to convert some photos to PNG format on my Mac. So far, the output files appear to be fine. However, I'm receiving some confusing warning messages that seem to conflict with the manual.

ImageMagick was installed through homebrew.

Some basic information:

```
$ convert --version
Version: ImageMagick 7.1.1-30 Q16-HDRI aarch64 22086 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(5.0) 
Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg jxl lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib zstd
Compiler: gcc (4.2)
```

```
macOS: 14.4.1
Mac: MacBook Pro M1 2020
```

The command I used:

```
convert input.tif -compress Zip -define png:compression-filter=5 -define png:compression-level=0 output.png
```

Output / Error (warning) message:

```
convert: ignoring invalid defined png:compression-filter =5 @ warning/png.c/WritePNGImage/11981.
```

I checked the [manual](https://imagemagick.org/script/defines.php) saying that:

> png:compression-filter=value
>
> Valid values are 0 through 9. 0-4 are the corresponding PNG filters, 5 means adaptive filtering except for images with a colormap, 6 means adaptive filtering for all images, 7 means MNG "loco" compression, 8 means Z_RLE strategy with adaptive filtering, and 9 means Z_RLE strategy with no filtering.

Value 5 should be considered valid, but the program indicates it as invalid. Additionally, the input pictures do not contain a colormap.

**Comment by dlemstra** [comment, marked as answer]
Thanks for reporting this. I just pushed a patch to make sure you no longer get this warning.
